### PR TITLE
feat: complete account linking workflow

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: src/model/registerConsent.model.ts [hunk 1]
```diff
@@ -112,17 +119,76 @@ export class RegisterConsentModel
...
+  }
+
+  async onStoreConsent (): Promise<void> {
+    const { consentsPostRequestAUTH, participantDFSPId } = this.data
+    try {
+      await createAndStoreConsent(
+        consentsPostRequestAUTH.consentId,
+        // initiatorId is deprecated so just using empty string for now
+        '',
+        // todo: double check if this is supposed to be the DFSP's id
```
Comment (lewisdaly) [inline src/model/registerConsent.model.ts:185]: Hmm I think so...

File: test/unit/model/registerConsent.model.test.ts [hunk 2]
```diff
@@ -58,89 +62,98 @@ jest.mock('~/shared/pub-sub')
...
-      accountId: 'dfsp.username.5678',
-      actions: [
-        'accounts.transfer',
-        'accounts.getBalance'
-      ]
-    }
+      accountId: 'f67170bc-3702-4786-90cd-023466b575ca',
+      // TODO: missing in payload for some reason on front end client...
+      // This isn't a big deal, as long as we derive the challenge in the same way.
+      // actions: [
```
Comment (lewisdaly) [inline test/unit/model/registerConsent.model.test.ts:79]: Is this fixed now with the newer payload I gave you?

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Looks good, just a couple minor nitpicks.

File: src/model/registerConsent.model.ts (hunk 1 above)
Comment (kleyow) [reply src/model/registerConsent.model.ts:185]: honestly theres still much to do on the auth-service
config needs to be refactored
old fido2 code cleanup
database schema changes maybe...

just trying to get this over the line

File: test/unit/model/registerConsent.model.test.ts (hunk 2 above)
Comment (kleyow) [reply test/unit/model/registerConsent.model.test.ts:79]: yep 🙃 I'll remove this comment in a bit

File: src/model/registerConsent.model.ts (hunk 1 above)
Comment (kleyow) [reply src/model/registerConsent.model.ts:185]: but yes I concur in that I believe it's the DFSP's id so I'll remove the comment

File: test/unit/model/registerConsent.model.test.ts (hunk 2 above)
Comment (lewisdaly) [reply test/unit/model/registerConsent.model.test.ts]: Cool

**Comment by elnyry-sam-k** [comment]
Thanks Kevin, excited to see this!

**Comment by kleyow** [comment]
@lewisdaly ping

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**